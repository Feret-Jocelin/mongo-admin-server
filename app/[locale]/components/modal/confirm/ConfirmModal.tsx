import { useTranslations } from 'next-intl';

import GenericModal from '@components/modal/generic/GenericModal';

import styles from './confirmModal.module.scss';

interface ConfirmModalProps {
  open: boolean;
  description: string;
  onConfirm: () => void;
  onClose: () => void;
}

const ConfirmModal = ({
  open,
  description,
  onConfirm,
  onClose,
}: ConfirmModalProps) => {
  const t = useTranslations();

  return (
    <GenericModal
      open={open}
      title={t('modal.confirm.title')}
      onClose={onClose}
      onValidate={onConfirm}
    >
      <div className={styles['confirm-modal']}>
        <p>{description}</p>
      </div>
    </GenericModal>
  );
};

export default ConfirmModal;
