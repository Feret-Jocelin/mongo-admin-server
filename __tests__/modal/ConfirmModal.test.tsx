/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import ConfirmModal from '@/app/[locale]/components/modal/confirm/ConfirmModal';
import { NextIntlClientProvider } from 'next-intl';

describe('ConfirmModal', () => {
  const locale = 'fr';
  const message = require(`../../shared/locales/${locale}.json`)
  it('renders', () => {
    const handleClose = jest.fn();
    const handleConfirm = jest.fn();
    const { container } = render(
      <NextIntlClientProvider locale={locale} messages={message}>
        <ConfirmModal 
          open={true}
          description='Common Confirm Modal'
          onClose={handleClose}
          onConfirm={handleConfirm} 
        />
      </NextIntlClientProvider>
    )
    expect(container).toMatchSnapshot();
  })
})