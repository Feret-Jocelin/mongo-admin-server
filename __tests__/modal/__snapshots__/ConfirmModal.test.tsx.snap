// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal renders 1`] = `
<div>
  <dialog
    class="modal"
  >
    <section
      class="modal__header"
    >
      <div
        class="modal__header--icon"
      >
        <div
          class="undefined pointer-cursor"
        />
      </div>
      <h2>
        Confirmation
      </h2>
    </section>
    <section
      class="modal__container"
    >
      <div
        class="confirm-modal"
      >
        <p>
          Common Confirm Modal
        </p>
      </div>
    </section>
    <section
      class="modal__footer"
    >
      <button
        class="button full outline center"
        type="button"
      >
        Annuler
      </button>
      <button
        class="button full center"
        type="button"
      >
        Confirmer
      </button>
    </section>
  </dialog>
</div>
`;
